bgcval2/analysis_timeseries.py

```diff
@@ -139,6 +163,180 @@ def build_list_of_suite_keys(suites, debug=True):
     return analysis_keys
 
 
+def load_function(functionname):
+    """
+    Using the named function in the key yaml, load that function and return it. 
+    """
+    # load functions:
+    if functionname in std_functions.keys():
+        print( "Standard Function Found:", functionname)
+        func = std_functions[functionname]
+
+    if functionname.find(':') > -1:
+        [functionFileName,functionname] = functionname.split(':')
+        lst = functionFileName.replace('.py','').replace('/', '.').split('.')
+        modulename =  '.'.join(lst)
+
+        print("parseFunction:\tAttempting to load the function:",functionname, "from the:",modulename)
+        mod = __import__(modulename, fromlist=[functionname,])
+        func = getattr(mod, functionname)
+    return func     
+
+
+def load_function_kwargs(dictionary, m_or_d):
+    """
+    Loads key word arguments from yaml file dictionairy.
+    """
+    data_kwargs = {}
+    #, data_convert_kwargss = [], []
+    #####
+    # Looking for kwargs to pass to convert:
+    for key, value in dictionary.items():
+        searchFor =  m_or_d+'_convert_'
+        findstr = key.find(searchFor)
+        if findstr==-1: continue
+        kwargkey = key[len(searchFor):]
+        data_kwargs[kwargkey] = value
+    return data_kwargs 
+
+
+def findReplaceFlag(filepath, flag, new_value):
+    """
+    Looking for $FLAG in the filepath.
+    If found, we replace $FLAG with new_value
+    """
+    lookingFor = '$'+flag.upper()
+    if filepath.find(lookingFor) ==-1: 
+        return filepath
+    filepath = filepath.replace(lookingFor, new_value)
+    return filepath
+
+
+def parse_list_from_string(list1):
+    """
+    This tool takes a string and returns it as a list.
+    """
+    if isinstance(list1, list):
+       return list1
+
+    list1 = list1.replace('\t', ' ')
+    list1 = list1.replace(',', ' ')
+    list1 = list1.replace('  ', ' ')
+    list1 = list1.replace('\'', '')
+    list1 = list1.replace('\"', '')
+    list1 = list1.replace(';', '')
+    while list1.count('  ')>0:
+        list1 = list1.replace('  ', ' ')
+    #if len(list1)==0: return []
+    return list1.split(' ')
```

Comment:
this whole thing can be done in one line: `list1.replace(',', ' ').replace('\'', '').replace(';', '').strip().split()`